extensions/amp-form/0.1/test-e2e/test-ssr.js

```diff
@@ -0,0 +1,50 @@
+/*e
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Amp-form test for standalone and in an SSR enabled viewer. Verifies that
+ * the rendered output is the same if the response from the REST endpoint is
+ * the same as from the SSR endpoint.
+ */
+
+describes.endtoend('AMP form server side rendered templates', {
+  testUrl: 'http://localhost:8000/test/fixtures/e2e/amp-form/amp-form.ssr.html',
+  environments: ['single', 'viewer-demo'],
+  experiments: ['layers'],
+}, async env => {
+  let controller;
+
+  beforeEach(async() => {
+    controller = env.controller;
+  });
+
+  it('should render server side render form response', async() => {
+    const searchInput =
+        await controller.findElement('#xhr-get  input[name=term]');
+    controller.type(searchInput, 'search term');
```

Comment:
You must always `await` controller methods<b>*</b>. Without `await`, execution will continue without waiting for typing to finish.

<b>*</b><sub>(with the exception of methods that return `ControllerPromise`s that get passed to `expect` for polling)</sub>